src/alert.cpp

```diff
@@ -1777,5 +1777,63 @@ namespace libtorrent {
 		return buf;
 	}
 
+	dht_get_peers_reply_alert::dht_get_peers_reply_alert(aux::stack_allocator& alloc
+		, sha1_hash const& ih
+		, std::vector<tcp::endpoint> const& peers)
+		: m_alloc(alloc)
+		, info_hash(ih)
+		, m_num_peers(peers.size())
+		, m_peers_idx(alloc.allocate(sizeof(boost::int32_t) * peers.size() * 2))
+	{
+		for (int i = 0; i < m_num_peers; i++) {
+			tcp::endpoint endp = peers[i];
+			boost::int32_t size = (boost::int32_t)endp.size();
+			boost::int32_t idx = (boost::int32_t)alloc.copy_buffer((char*)endp.data(), size);
+
+			char* idx_ptr = alloc.ptr(m_peers_idx) + sizeof(boost::int32_t) * (2 * i);
+			char* size_ptr = idx_ptr + sizeof(boost::int32_t);
+
+			detail::write_int32(idx, idx_ptr);
+			detail::write_int32(size, size_ptr);
+		}
+	}
```

Comment:
oh, I see now. You make multiple allocations here. Now, I was expecting this code to allocate a single array, then copy all the peers into it. This seems more complex than it needs to be. If the point is to support IPv6 (which actually isn't supported by the DHT), I think it would be simpler to just allocate two arrays, one for IPv4 and one for IPv6. and instead of taking the endpoint directly, you could pull out the address_v4::to_bytes() and uint16_t for the port.